deepinv/tests/test_datasets.py

```diff
@@ -202,8 +203,111 @@ def test_base_dataset():
             check_dataset(MyDataset(bad_dataset_input))
 
 
+SPLIT_NAMES = ["train", "test", "val", "dummy"]
+
+
+@pytest.mark.parametrize("train", [True, False])
+@pytest.mark.parametrize("split", [*SPLIT_NAMES, None])
+@pytest.mark.parametrize("with_transform", [False, True])
+@pytest.mark.parametrize("load_physics_generator_params", [True, False])
+@pytest.mark.parametrize("dtype", [torch.float32, torch.float64])
+@pytest.mark.parametrize("complex_dtype", [torch.complex64, torch.complex128])
+@pytest.mark.parametrize("length", [10])
+@pytest.mark.parametrize("with_params", [False, True])
+@pytest.mark.parametrize("unsupervised", [False, True])
+@pytest.mark.parametrize("close", [False, True])
+@pytest.mark.parametrize("stack_size", [1, 2, 3])
+def test_hdf5dataset(
```

Comment:
should we also have an end-to-end dataset test showing all the possibilities of data you can save and load (stacked, params, unsupervised etc.), and simply test that the input dataset -> save -> load = output dataset?

Reply:
I guess it would become a superset of the test_hdf5dataset_generate_dataset yes. but we can keep them separate for now, it would only add minimal overhead to do what I suggested